### PR TITLE
Add async filtering to portfolios

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- Added async filtering by name for portfolios
- Portfolios component is now function using hooks
- updated react-router-dom to be able to use router hooks

### Fixes
- fixed default count value for async pagination to 0
  - no longer shows 0 - 50 before the first API request is finished. Now its 0 - 0

### CODE REVIEW

File: src/smart-components/portfolio/portfolios.js [hunk 1]
```diff
@@ -17,102 +15,90 @@ import { fetchPortfolios } from '../../redux/actions/portfolio-actions';
 import PortfolioCard from '../../presentational-components/portfolio/porfolio-card';
 import createPortfolioToolbarSchema from '../../toolbar/schemas/portfolios-toolbar.schema';
 import ContentGalleryEmptyState, { EmptyStatePrimaryAction } from '../../presentational-components/shared/content-gallery-empty-state';
+import asyncFormValidator from '../../utilities/async-form-validator';
+
+const debouncedFilter = asyncFormValidator((value, dispatch, filteringCallback) => {
+  filteringCallback(true);
+  dispatch(fetchPortfolios(value, defaultSettings)).then(() => filteringCallback(false));
```
Comment (lgalis) [inline src/smart-components/portfolio/portfolios.js]: - any filtering will reset the pagination to the default options.
maybe use the pagination option here:
const debouncedFilter = asyncFormValidator((value, dispatch, filteringCallback, pagination) => {
  filteringCallback(true);
  dispatch(fetchPortfolios(value, pagination)).then(() => filteringCallback(false));
}, 1000);


and then:
-------
 const handleFilterItems = value => {
    stateDispatch({ type: 'setFilterValue', payload: value });
    debouncedFilter(value, dispatch, isFiltering => stateDispatch({ type: 'setFilteringFlag', payload: isFiltering }), meta);
  };

Comment (gmcculloug) [reply src/smart-components/portfolio/portfolios.js]: @lgalis Is that the same as the bug I mentioned this morning? 
 https://projects.engineering.redhat.com/browse/SSP-630

Comment (Hyperkid123) [reply src/smart-components/portfolio/portfolios.js]: @lgalis we can only send limit not offset. Offset has to be reset but you are right that limit can be kept.

Comment (Hyperkid123) [reply src/smart-components/portfolio/portfolios.js]: > @lgalis Is that the same as the bug I mentioned this morning?
> https://projects.engineering.redhat.com/browse/SSP-630

It is the same.

**Comment by Hyperkid123** [comment]
@lgalis @gmcculloug limit should be now kept when filtering.

File: src/smart-components/portfolio/portfolios.js (hunk 1 above)
Comment (lgalis) [reply src/smart-components/portfolio/portfolios.js]: @Hyperkid123 - sounds good.

**Pull request merged**